### ISSUE TITLE
Few-shot Image Generation with Elastic Weight Consolidation

### ISSUE THREAD

**Issue by nmhkahn** [nmhkahn/paper-notes#8]
![](https://user-images.githubusercontent.com/16869368/129996794-dca93ae6-8079-4210-bc7e-10be0976f248.png)
- Large source domain에서 학습한 G 모델을 small target domain 으로 transfer
- 분석해보니 weight들은 서로다른 importance를 가지더라 -> adaptation 할때 모든 weight들을 동일선상에 놓고 학습하면 안됨
    - Weight 전체를 fine-tune하면 당연하게도 overfit 발생 우려 높아짐
- 그래서 1) 어떤 weight이 중요한지, 혹은 변경하면 안되는지 변경해도 되는지를 판단하고 2) 이 판단 근거를 토대로 loss term에 reg effect 줘야함

![](https://user-images.githubusercontent.com/16869368/129997098-2a23a65e-8c62-4dfa-ae5b-6f55cbd9fb38.png)
- Face->emoji로 분석해보니까 마지막 레이어쪽은 fine-tune 안하는게 나음
- ![](https://user-images.githubusercontent.com/16869368/129997220-d0db7d06-8697-454c-8b8b-701f7e97f1a7.png)
- Importance measure를 위해 Fisher Information을 사용함. 위 분석 결과랑 비슷한 observation이 나와서 이거 사용해도 무방함
    - (그런데, 다른 도메인에서도 마찬가지 결과인지는 잘 모르겠음)
- Classification task에서 Elastic Weight Consolidation loss라는게 Fisher Information 기반으로 있었는데, 우리 이거 그대로 가져다가 reg term으로 사용함
- ![](https://user-images.githubusercontent.com/16869368/129997422-bc9b457c-361e-4b10-8e5d-261e635d47a7.png)
- 이거 쓰면 잘된다 그런데 아래 그림처럼 target domain이 많이 다르면 EWC 사용해도 잘 안됨
![](https://user-images.githubusercontent.com/16869368/129997506-341f20d0-9f4b-4eec-8dc4-434e5a9f42d0.png)
